### ISSUE TITLE
/ollama/v1/chat/completions endpoint broke (405 Method Not Allowed) in v0.3.6

### ISSUE THREAD

**Issue by kobaltz** [open-webui/open-webui#3489]
# Bug Report

## Description

**Bug Summary:**
Making a HTTP POST to /ollama/v1/chat/completions worked in v0.3.5 but now returns 405 - Method not allowed

**Steps to Reproduce:**
Using the docker compose file which at the time was referencing v0.3.6, I was unable to make API calls to the Chat Completion
Reverting the open webui image to v0.3.5 and the POST works as expected.
```
services:
  ollama:
    volumes:
      - ollama:/root/.ollama
    container_name: ollama
    pull_policy: always
    tty: true
    restart: unless-stopped
    image: ollama/ollama:${OLLAMA_DOCKER_TAG-latest}
    environment:
      - NVIDIA_VISIBLE_DEVICES=all
      - NVIDIA_DRIVER_CAPABILITIES=all
    deploy:
      resources:
        reservations:
          devices:
            - capabilities: [gpu]
  open-webui:
    build:
      context: .
      args:
        OLLAMA_BASE_URL: '/ollama'
      dockerfile: Dockerfile
    image: ghcr.io/open-webui/open-webui:${WEBUI_DOCKER_TAG-main}
    container_name: open-webui
    volumes:
      - open-webui:/app/backend/data
    depends_on:
      - ollama
    ports:
      - ${OPEN_WEBUI_PORT-2039}:8080
    environment:
      - 'OLLAMA_BASE_URL=http://ollama:11434'
    extra_hosts:
      - host.docker.internal:host-gateway
    restart: unless-stopped

volumes:
  ollama: {}
  open-webui: {}
```

**Expected Behavior:**
HTTP POST to /ollama/v1/chat/completions returns the JSON response

**Actual Behavior:**
405 Method Not Allowed when making HTTP POST to /ollama/v1/chat/completions

## Environment

- **Open WebUI Version:** main (and v0.3.6)
- **Ollama (if applicable):** latest (and 0.1.47)

- **Operating System:** Debian Bookworm
- **Browser (if applicable):** n/a

## Reproduction Details

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the Docker container logs.

## Logs and Screenshots

**Docker Container Logs:**
```
INFO:apps.ollama.main:url: http://ollama:11434
INFO:     my-ip:0 - "POST /ollama/v1/chat/completions HTTP/1.1" 405 Method Not Allowed
INFO:apps.ollama.main:url: http://ollama:11434
INFO:     my-ip:0 - "POST /ollama/v1/chat/completions HTTP/1.1" 405 Method Not Allowed
```

## Installation Method

Docker Compose

**Comment by debemdeboas** [comment]
Seeing this here as well.
Also seeing a `TypeError: signup() missing 1 required positional argument: 'form_data'`.
Going back to v0.3.5 fixes things.

**Comment by justinh-rahb** [comment]
Can you share the exact cURL command you've used to test, as well as additional browser dev console logs, maybe some more backend logs as well to help us out.

**Comment by kobaltz** [comment]
Here's the cURL command. 

```
curl -X POST \
  http://127.0.0.1:2039/ollama/v1/chat/completions \
  -H 'Content-Type: application/json' \
  -H 'Authorization: Bearer sk-89ddf4d66bae4527974150ee796e71f6' \
  -d '{
       "model": "llama3:instruct",
       "messages": [
         {
           "role": "user",
           "content": "Tell me a story"
         }
       ],
       "max_tokens": 50
     }'
     
=> {"detail":"Method Not Allowed"}
```

But, if I change the docker compose file to `v0.3.5` and run the same command, I get

```
=> {"id":"chatcmpl-138","object":"chat.completion","created":1719588557,"model":"llama3:instruct","system_fingerprint":"fp_ollama","choices":[{"index":0,"message":{"role":"assistant","content":"Once upon a time, in a small village nestled in the rolling hills of Tuscany, there was a little girl named Sophia. She lived with her family in a charming stone house with a bright red door and a garden full of fragrant herbs"},"finish_reason":"length"}],"usage":{"prompt_tokens":0,"completion_tokens":50,"total_tokens":50}}
```

Since I'm using just the cURL command, there isn't any additional logs as the docker container logas are just saying that the method isn't allowed. Maybe there is a debug ENV var I could add to get a more verbose output?

**Comment by abishekmuthian** [comment]
I'm getting `Method Not Allowed`  for `deepseek-coder-v2:latest ` model in v0.3.6. Other models work fine.

`INFO:     127.0.0.1:41212 - "POST /ollama/api/chat HTTP/1.1" 405 Method Not Allowed`

**Comment by odv-ops** [comment]
I'm getting `Method Not Allowed` for any models in v0.3.6
I used local k8s cluster, which was created with [kind](https://kind.sigs.k8s.io/) I got `Server connection verified
` for the Ollama API url `ollama.ollama.svc.cluster.local:11434`

```
INFO:apps.ollama.main:app.state.config.OLLAMA_BASE_URLS: ['http://ollama.ollama.svc.cluster.local:11434']
INFO:     127.0.0.1:56508 - "POST /ollama/urls/update HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
INFO:apps.ollama.main:get_all_models()
INFO:apps.ollama.main:get_all_models()
INFO:apps.openai.main:get_all_models()
INFO:     127.0.0.1:56508 - "POST /openai/keys/update HTTP/1.1" 200 OK
INFO:     127.0.0.1:56494 - "GET /ollama/api/version HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
INFO:apps.ollama.main:get_all_models()
INFO:     127.0.0.1:56508 - "GET /api/models HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
INFO:apps.ollama.main:get_all_models()
INFO:     127.0.0.1:56494 - "GET /api/models HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
INFO:apps.ollama.main:get_all_models()
INFO:apps.ollama.main:get_all_models()
INFO:     127.0.0.1:56546 - "GET /ollama/api/version HTTP/1.1" 200 OK
INFO:     127.0.0.1:56552 - "GET /api/v1/users/user/settings HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
INFO:     127.0.0.1:56562 - "GET /api/v1/users/user/settings HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
INFO:apps.ollama.main:get_all_models()
INFO:apps.ollama.main:url: http://ollama.ollama.svc.cluster.local:11434
INFO:     127.0.0.1:52376 - "POST /ollama/api/pull/0 HTTP/1.1" 200 OK
INFO:apps.ollama.main:get_all_models()
INFO:apps.ollama.main:get_all_models()
INFO:     127.0.0.1:52376 - "GET /api/models HTTP/1.1" 200 OK
INFO:     127.0.0.1:57818 - "POST /api/v1/chats/new HTTP/1.1" 200 OK
INFO:     127.0.0.1:57818 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
INFO:apps.ollama.main:url: http://ollama.ollama.svc.cluster.local:11434
INFO:     127.0.0.1:57818 - "POST /ollama/api/chat HTTP/1.1" 405 Method Not Allowed
ERROR:asyncio:Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0xfffe9473a910>
INFO:     127.0.0.1:57818 - "GET /api/v1/chats/ HTTP/1.1" 200 OK
```

**Comment by odv-ops** [comment]
Issue was resolved - there was a memory shortage and the OOM kicked in.

**Comment by tjbck** [comment]
Fixed on dev, 0.3.7 coming soon!

**Comment by Lanhild** [comment]
Commenting about this, I am getting an error `500` with the completions endpoint:

```
Sep 04 11:43:23 INFO: 173.178.45.182:0 - "POST /ollama/v1/chat/completions HTTP/1.1" 500 Internal Server Error
Sep 04 11:43:23 ERROR: Exception in ASGI application
Sep 04 11:43:23 + Exception Group Traceback (most recent call last):
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/_utils.py", line 87, in collapse_excgroups
Sep 04 11:43:23 | yield
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 190, in __call__
Sep 04 11:43:23 | async with anyio.create_task_group() as task_group:
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/anyio/_backends/_asyncio.py", line 680, in __aexit__
Sep 04 11:43:23 | raise BaseExceptionGroup(
Sep 04 11:43:23 | exceptiongroup.ExceptionGroup: unhandled errors in a TaskGroup (1 sub-exception)
Sep 04 11:43:23 +-+---------------- 1 ----------------
Sep 04 11:43:23 | Traceback (most recent call last):
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/uvicorn/protocols/http/httptools_impl.py", line 401, in run_asgi
Sep 04 11:43:23 | result = await app( # type: ignore[func-returns-value]
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
Sep 04 11:43:23 | return await self.app(scope, receive, send)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/fastapi/applications.py", line 1054, in __call__
Sep 04 11:43:23 | await super().__call__(scope, receive, send)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/applications.py", line 123, in __call__
Sep 04 11:43:23 | await self.middleware_stack(scope, receive, send)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 186, in __call__
Sep 04 11:43:23 | raise exc
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 164, in __call__
Sep 04 11:43:23 | await self.app(scope, receive, _send)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 189, in __call__
Sep 04 11:43:23 | with collapse_excgroups():
Sep 04 11:43:23 | File "/usr/lib/python3.10/contextlib.py", line 153, in __exit__
Sep 04 11:43:23 | self.gen.throw(typ, value, traceback)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/_utils.py", line 93, in collapse_excgroups
Sep 04 11:43:23 | raise exc
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 191, in __call__
Sep 04 11:43:23 | response = await self.dispatch_func(request, call_next)
Sep 04 11:43:23 | File "/app/code/backend/main.py", line 792, in update_embedding_function
Sep 04 11:43:23 | response = await call_next(request)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 165, in call_next
Sep 04 11:43:23 | raise app_exc
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 151, in coro
Sep 04 11:43:23 | await self.app(scope, receive_or_disconnect, send_no_error)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 189, in __call__
Sep 04 11:43:23 | with collapse_excgroups():
Sep 04 11:43:23 | File "/usr/lib/python3.10/contextlib.py", line 153, in __exit__
Sep 04 11:43:23 | self.gen.throw(typ, value, traceback)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/_utils.py", line 93, in collapse_excgroups
Sep 04 11:43:23 | raise exc
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 191, in __call__
Sep 04 11:43:23 | response = await self.dispatch_func(request, call_next)
Sep 04 11:43:23 | File "/app/code/backend/main.py", line 783, in check_url
Sep 04 11:43:23 | response = await call_next(request)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 165, in call_next
Sep 04 11:43:23 | raise app_exc
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 151, in coro
Sep 04 11:43:23 | await self.app(scope, receive_or_disconnect, send_no_error)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 189, in __call__
Sep 04 11:43:23 | with collapse_excgroups():
Sep 04 11:43:23 | File "/usr/lib/python3.10/contextlib.py", line 153, in __exit__
Sep 04 11:43:23 | self.gen.throw(typ, value, traceback)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/_utils.py", line 93, in collapse_excgroups
Sep 04 11:43:23 | raise exc
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 191, in __call__
Sep 04 11:43:23 | response = await self.dispatch_func(request, call_next)
Sep 04 11:43:23 | File "/app/code/backend/main.py", line 769, in commit_session_after_request
Sep 04 11:43:23 | response = await call_next(request)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 165, in call_next
Sep 04 11:43:23 | raise app_exc
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 151, in coro
Sep 04 11:43:23 | await self.app(scope, receive_or_disconnect, send_no_error)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/cors.py", line 85, in __call__
Sep 04 11:43:23 | await self.app(scope, receive, send)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 189, in __call__
Sep 04 11:43:23 | with collapse_excgroups():
Sep 04 11:43:23 | File "/usr/lib/python3.10/contextlib.py", line 153, in __exit__
Sep 04 11:43:23 | self.gen.throw(typ, value, traceback)
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/_utils.py", line 93, in collapse_excgroups
Sep 04 11:43:23 | raise exc
Sep 04 11:43:23 | File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 191, in __call__
Sep 04 11:43:23 | response = await self.dispatch_func(request, call_next)
Sep 04 11:43:23 | File "/app/code/backend/main.py", line 736, in dispatch
Sep 04 11:43:23 | content={"detail": e.args[1]},
Sep 04 11:43:23 | IndexError: tuple index out of range
Sep 04 11:43:23 +------------------------------------
Sep 04 11:43:23 2024-09-04T15:43:23.000Z
Sep 04 11:43:23 During handling of the above exception, another exception occurred:
Sep 04 11:43:23 2024-09-04T15:43:23.000Z
Sep 04 11:43:23 Traceback (most recent call last):
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/uvicorn/protocols/http/httptools_impl.py", line 401, in run_asgi
Sep 04 11:43:23 result = await app( # type: ignore[func-returns-value]
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
Sep 04 11:43:23 return await self.app(scope, receive, send)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/fastapi/applications.py", line 1054, in __call__
Sep 04 11:43:23 await super().__call__(scope, receive, send)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/applications.py", line 123, in __call__
Sep 04 11:43:23 await self.middleware_stack(scope, receive, send)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 186, in __call__
Sep 04 11:43:23 raise exc
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/errors.py", line 164, in __call__
Sep 04 11:43:23 await self.app(scope, receive, _send)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 189, in __call__
Sep 04 11:43:23 with collapse_excgroups():
Sep 04 11:43:23 File "/usr/lib/python3.10/contextlib.py", line 153, in __exit__
Sep 04 11:43:23 self.gen.throw(typ, value, traceback)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/_utils.py", line 93, in collapse_excgroups
Sep 04 11:43:23 raise exc
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 191, in __call__
Sep 04 11:43:23 response = await self.dispatch_func(request, call_next)
Sep 04 11:43:23 File "/app/code/backend/main.py", line 792, in update_embedding_function
Sep 04 11:43:23 response = await call_next(request)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 165, in call_next
Sep 04 11:43:23 raise app_exc
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 151, in coro
Sep 04 11:43:23 await self.app(scope, receive_or_disconnect, send_no_error)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 189, in __call__
Sep 04 11:43:23 with collapse_excgroups():
Sep 04 11:43:23 File "/usr/lib/python3.10/contextlib.py", line 153, in __exit__
Sep 04 11:43:23 self.gen.throw(typ, value, traceback)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/_utils.py", line 93, in collapse_excgroups
Sep 04 11:43:23 raise exc
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 191, in __call__
Sep 04 11:43:23 response = await self.dispatch_func(request, call_next)
Sep 04 11:43:23 File "/app/code/backend/main.py", line 783, in check_url
Sep 04 11:43:23 response = await call_next(request)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 165, in call_next
Sep 04 11:43:23 raise app_exc
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 151, in coro
Sep 04 11:43:23 await self.app(scope, receive_or_disconnect, send_no_error)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 189, in __call__
Sep 04 11:43:23 with collapse_excgroups():
Sep 04 11:43:23 File "/usr/lib/python3.10/contextlib.py", line 153, in __exit__
Sep 04 11:43:23 self.gen.throw(typ, value, traceback)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/_utils.py", line 93, in collapse_excgroups
Sep 04 11:43:23 raise exc
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 191, in __call__
Sep 04 11:43:23 response = await self.dispatch_func(request, call_next)
Sep 04 11:43:23 File "/app/code/backend/main.py", line 769, in commit_session_after_request
Sep 04 11:43:23 response = await call_next(request)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 165, in call_next
Sep 04 11:43:23 raise app_exc
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 151, in coro
Sep 04 11:43:23 await self.app(scope, receive_or_disconnect, send_no_error)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/cors.py", line 85, in __call__
Sep 04 11:43:23 await self.app(scope, receive, send)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 189, in __call__
Sep 04 11:43:23 with collapse_excgroups():
Sep 04 11:43:23 File "/usr/lib/python3.10/contextlib.py", line 153, in __exit__
Sep 04 11:43:23 self.gen.throw(typ, value, traceback)
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/_utils.py", line 93, in collapse_excgroups
Sep 04 11:43:23 raise exc
Sep 04 11:43:23 File "/usr/local/lib/python3.10/dist-packages/starlette/middleware/base.py", line 191, in __call__
Sep 04 11:43:23 response = await self.dispatch_func(request, call_next)
Sep 04 11:43:23 File "/app/code/backend/main.py", line 736, in dispatch
Sep 04 11:43:23 content={"detail": e.args[1]},
Sep 04 11:43:23 IndexError: tuple index out of range
```
